apps/todos/src/app/todos/todos.component.spec.ts

```diff
@@ -84,9 +95,244 @@ describe('TodosComponent', () => {
+          },
+        ],
+        excludeComponentDeclaration: true,
+      });
+
+      expect(screen.getByRole('heading')).toHaveTextContent('Test Heading');
+    });
+  });
+
+  describe('using testing-library and ts-mockito (shallow)', () => {
+    it('should query todo items', async () => {
+      let mockTodosService = mock(TodosService); // has type TodosService
+      let mockActivatedRoute = mock(ActivatedRoute); // has type ActivatedRoute
+
+      when(mockTodosService.query(anyString())).thenReturn(of([]));
+      when(mockActivatedRoute.paramMap).thenReturn(
+        of(
+          convertToParamMap({
+            query: 'all',
+          })
+        )
+      );
+      await render(TodosComponent, {
+        providers: [
+          {
+            provide: TodosService,
+            useFactory: () => instance(mockTodosService), // has type TodosService
+          },
+          {
+            provide: ActivatedRoute,
+            useFactory: () => instance(mockActivatedRoute),
+          },
+        ],
+        schemas: [CUSTOM_ELEMENTS_SCHEMA],
+      });
+
+      verify(mockTodosService.query('all')).called();
+
+      const [param] = capture(mockTodosService.query).first();
+      expect(param).toEqual('all');
+    });
+
+    // it('should show app title') is the same as above
+  });
+
+  describe('using testing-library and ng-mockito (shallow)', () => {
+    it('should query todo items', async () => {
+      let mockTodosService = mock(TodosService); // has type TodosService
+
+      await render(TodosComponent, {
+        providers: [
+          mockProvider(mockTodosService, (mock) =>
+            when(mock.query(anyString())).thenReturn(of([]))
+          ),
+          mockProvider(ActivatedRoute, (mock) =>
+            when(mock.paramMap).thenReturn(
+              of(
+                convertToParamMap({
+                  query: 'all',
+                })
+              )
+            )
+          ),
+        ],
+        schemas: [CUSTOM_ELEMENTS_SCHEMA],
+      });
+
+      verify(mockTodosService.query('all')).called();
+
+      const [param] = capture(mockTodosService.query).first();
+      expect(param).toEqual('all');
+    });
+
+    it('should show app title', async () => {
+      let mockTodosService = mock(TodosService); // has type TodosService
+
+      await render(TodosComponent, {
+        providers: [
+          mockNg(mockTodosService, (mock) =>
+            when(mock.query(anyString())).thenReturn(NEVER)
+          ),
+          mockNg(ActivatedRoute, (mock) =>
+            when(mock.paramMap).thenReturn(NEVER)
+          ),
+          mockNg([APP_TITLE, TodosComponent], {
+            use: 'Test Heading' /* type-safe */,
+          }),
+        ],
+        schemas: [CUSTOM_ELEMENTS_SCHEMA],
+      });
+
+      expect(screen.getByRole('heading')).toHaveTextContent('Test Heading');
+    });
+  });
+
+  describe('using ts-mockito (isolated)', () => {
```

Comment:
Idee, die jeweiligen setups/ansätze können wir in jeweils eine Datei auslagern.

Reply:
ja hatte ich auch überlegt und war zu faul ;-)